app/scripts/modules/core/src/pipeline/config/stages/googleCloudBuild/GoogleCloudBuildStageForm.tsx

```diff
@@ -1,234 +1,220 @@
-        {stage.buildDefinitionSource === buildDefinitionSources.TEXT && (
-          <StageConfigField label="Build Definition">
-            <YamlEditor value={this.state.rawBuildDefinitionYaml} onChange={this.onYamlChange} />
-          </StageConfigField>
-        )}
-        {stage.buildDefinitionSource === buildDefinitionSources.ARTIFACT && SETTINGS.feature['artifactsRewrite'] && (
-          <StageConfigField label="Build Definition Artifact">
-            <StageArtifactSelector
-              artifact={get(stage, 'buildDefinitionArtifact.artifact')}
-              excludedArtifactTypePatterns={[
-                ArtifactTypePatterns.DOCKER_IMAGE,
-                ArtifactTypePatterns.KUBERNETES,
-                ArtifactTypePatterns.FRONT50_PIPELINE_TEMPLATE,
-                ArtifactTypePatterns.EMBEDDED_BASE64,
-              ]}
-              expectedArtifactId={get(stage, 'buildDefinitionArtifact.artifactId')}
-              onArtifactEdited={this.setArtifact}
-              onExpectedArtifactSelected={(artifact: IExpectedArtifact) => this.setArtifactId(artifact.id)}
-              pipeline={this.props.pipeline}
-              stage={stage}
-            />
-          </StageConfigField>
-        )}
-        {stage.buildDefinitionSource === buildDefinitionSources.ARTIFACT && !SETTINGS.feature['artifactsRewrite'] && (
-          <PreRewriteStageArtifactSelector
-            excludedArtifactTypePatterns={[
-              ArtifactTypePatterns.DOCKER_IMAGE,
-              ArtifactTypePatterns.KUBERNETES,
-              ArtifactTypePatterns.FRONT50_PIPELINE_TEMPLATE,
-              ArtifactTypePatterns.EMBEDDED_BASE64,
-            ]}
-            pipeline={this.props.pipeline}
-            selectedArtifactAccount={get(stage, 'buildDefinitionArtifact.artifactAccount')}
-            selectedArtifactId={get(stage, 'buildDefinitionArtifact.artifactId')}
-            setArtifactAccount={this.setArtifactAccount}
-            setArtifactId={this.setArtifactId}
-            stage={stage}
-            updatePipeline={this.props.updatePipeline}
+            isLoading={fetchAccountsStatus === 'PENDING'}
+            stringOptions={fetchAccountsResult}
           />
         )}
-        {stage.buildDefinitionSource === buildDefinitionSources.TRIGGER && (
-          <>
-            <StageConfigField label="Trigger Name">
-              <Select
+      />
+      <FormikFormField
+        fastField={false}
+        label="Build Definition Source"
+        name="buildDefinitionSource"
+        input={(inputProps: IFormInputProps) => <RadioButtonInput {...inputProps} options={SOURCE_OPTIONS} />}
+      />
+      {stage.buildDefinitionSource === BuildDefinitionSource.TEXT && (
+        <FormikFormField
+          fastField={false}
+          label="Build Definition"
+          name="buildDefinition"
+          input={(inputProps: IFormInputProps) => (
+            <YamlEditor {...inputProps} value={rawBuildDefinitionYaml} onChange={onYamlChange} />
```

Comment:
it might be worth wrapping `YamlEditor` as a `FormInput` compatible component `YamlEditorInput` in some future PR.  It would be nice if each form that uses yaml didn't have to know how to manage single/multi-document yaml.

Reply:
Yes, absolutely -- will do this as part of converting the Kubernetes stages that also use a YamlEditor.